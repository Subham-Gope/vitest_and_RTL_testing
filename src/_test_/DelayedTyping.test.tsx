// import { describe, test, expect } from "vitest";
// import { render, screen } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
// import DelayedTyping from "../components/DelayedTyping";

// describe("Testing the DelayedTyping.tsx file", () => {
//   test("should type with a delay in keystroke", async () => {
//     const user = userEvent.setup();

//     const { container } = render(<DelayedTyping message={"Hello"} />);

//     expect(container).toMatchSnapshot();

//     const input = screen.getByRole("textbox", { name: /delayed/i });
//     expect(input).toBeInTheDocument();

//     // initiating user event
//     input.focus();
//     await user.type(input, "Delayed", { delay: 300, skipClick: true });

//     expect(input).toHaveValue("Delayed");
//     const displayText = screen.getByText(/input text/i);
//     expect(displayText).toBeInTheDocument();
//   });
// });
