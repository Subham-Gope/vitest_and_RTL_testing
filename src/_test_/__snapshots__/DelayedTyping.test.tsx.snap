// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing the DelayedTyping.tsx file > should type with a delay in keystroke 1`] = `
<div>
  <h1>
    Hello
  </h1>
  <div>
    <label
      for="delayed-typing"
    >
      Delayed Typing Test
    </label>
    <input
      id="delayed-typing"
      placeholder="Enter your text"
      type="text"
      value=""
    />
  </div>
</div>
`;
